### DISCUSSION TITLE
Does x-teleport.replace (or equivalent) exist?

### DISCUSSION THREAD

**Discussion by gjcourtney** [alpinejs/alpine#4223] [1. Help]
I've got a scenario where I've got an Alpine component making a `fetch()` request to get some HTML back from the server. Within my component, I'm using `x-html` to bind the returned HTML content into the UI where I want it. All good.

I would however, also like to update another section of my UI with some of the content that comes back in the HTML response. This is a totally separate fragment of my page, so isn't within the scope of the Alpine component that made the initial `fetch()` request. I could extract the DOM nodes from the HTML response and then `$dispatch('update-my-fragment', {html: domNodes})` to send the HTML to another Alpine component, but that feels more clunky and labour intensive than I'd ideally like.

What ocurred to me was that I'm essentially looking to teleport the HTML to another part of my page. In my head I want to do something like `x-teleport.replace="#fragment"`, however x-teleport appears only to be capable of appending and not replacing.

Before I work out how to raise a PR for the functionality I'm looking for, I wondered if others had an existing approach to what I'm describing, and also whether a PR for this is likely to be accepted?

**Comment by ekwoka** [comment]
So what is wrong with the appending in that case though?

You could also, since you have it, manually do it.

```js
document.querySelector('#fragment').replaceWith(domstring)
```

**Reply by gjcourtney** [reply]
In a few instances within our application, we need to refresh the contents of the Alpine component multiple times based on lifecycle events, which will in turn append the HTML to the fragment multiple times, resulting in duplicated content.

Regarding doing it manually, yes, that's possible, but requires me to put the same logic into my codebase multiple times. If `x-teleport.replace` was supported, that would be ideal. Having [looked at the x-teleport code](https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-teleport.js#L36), it looks like it would be a relatively simple addition to support the modifier, but before I raised the PR, I wanted to check if I was missing something and whether it's likely to be accepted.

**Reply by SimoTod** [reply]
With replace, you are going to lose the target and it could break the morph plugin (and livewire as well). Probably okay for your use case but if it breaks morph, it won't be merged in so you need to verify that.

**Reply by gjcourtney** [reply]
Cool, thanks for the tip about morph, wasn't aware of that. I might see if I can do something with it along the lines of `<div x-init="Alpine.morph(document.getElementById('#fragment'), $el.outerHtml)">...</div>`

**Reply by ekwoka** [reply]
That should work fine.

You could also make a custom directive, or simple function.

x-teleport.replace will still need to be written multiple times.

**Reply by walirazzaq** [reply]
@SimoTod I also came across somewhat similar use case where I needed to replace the contents of the `target` not replacing the whole `target`.  Maybe an `x-teleport.overwrite` modifier would helpful another `else if` condition [here](https://github.com/alpinejs/alpine/blob/541cb8ef8fb169cdb6032fdf920169ffd083c812/packages/alpinejs/src/directives/x-teleport.js#L43)  and doing it like:
```JS
{
target.innerHTML = ''
target.appendChild(node)
}
```
Not sure if I am missing something because I don't have extensive knowledge about Alpines' morph.

**Reply by ekwoka** [reply]
```js
target.replaceChildren(node)
```
